system_tests/logging_.py

```diff
@@ -81,28 +64,26 @@ def setUp(self):
 
     def tearDown(self):
         from gcloud.exceptions import NotFound
+        retry = RetryErrors(NotFound)
         for doomed in self.to_delete:
-            backoff_intervals = [1, 2, 4, 8]
-            while True:
-                try:
-                    doomed.delete()
-                    break
-                except NotFound:
-                    if backoff_intervals:
-                        time.sleep(backoff_intervals.pop(0))
-                    else:
-                        raise
+            retry(doomed.delete)()
 
     @staticmethod
     def _logger_name():
         return 'system-tests-logger' + unique_resource_id('-')
 
+    def _list_entries(self, logger):
+        from grpc.framework.interfaces.face.face import AbortionError
+        inner = RetryResult(_has_entries)(logger.list_entries)
+        outer = RetryErrors(AbortionError, _retry_on_unavailable)(inner)
+        return outer()
```

Comment:
This sucks we have to deal with eventual consistency and with connection failures